core/src/test/scala/integration/kafka/admin/RemoteTopicCrudTest.scala

```diff
@@ -0,0 +1,331 @@
+          AlterConfigOp.OpType.SET),
+        new AlterConfigOp(new ConfigEntry(TopicConfig.LOCAL_LOG_RETENTION_BYTES_CONFIG, "100"),
+          AlterConfigOp.OpType.SET)
+      ))
+    admin.incrementalAlterConfigs(configs).all().get()
+    verifyRemoteLogTopicConfigs(topicConfig)
+  }
+
+  @ParameterizedTest(name = TestInfoUtils.TestWithParameterizedQuorumName)
+  @ValueSource(strings = Array("zk", "kraft"))
+  def testUpdateTopicConfigWithInheritedLocalRetentionTime(quorum: String): Unit = {
+    val admin = createAdminClient()
+    val topicConfig = new Properties()
+    topicConfig.put(TopicConfig.REMOTE_LOG_STORAGE_ENABLE_CONFIG, "true")
+    TestUtils.createTopicWithAdmin(admin, testTopicName, brokers, numPartitions, numReplicationFactor,
+      topicConfig = topicConfig)
+
+    // inherited local retention ms is 1000
+    val configs = new util.HashMap[ConfigResource, util.Collection[AlterConfigOp]]()
+    configs.put(new ConfigResource(ConfigResource.Type.TOPIC, testTopicName),
+      util.Arrays.asList(
+        new AlterConfigOp(new ConfigEntry(TopicConfig.RETENTION_MS_CONFIG, "200"),
+          AlterConfigOp.OpType.SET),
+      ))
+    assertThrowsException(classOf[InvalidConfigurationException],
+      () => admin.incrementalAlterConfigs(configs).all().get())
+  }
+
+  @ParameterizedTest(name = TestInfoUtils.TestWithParameterizedQuorumName)
+  @ValueSource(strings = Array("zk", "kraft"))
+  def testUpdateTopicConfigWithInheritedLocalRetentionSize(quorum: String): Unit = {
+    val admin = createAdminClient()
+    val topicConfig = new Properties()
+    topicConfig.put(TopicConfig.REMOTE_LOG_STORAGE_ENABLE_CONFIG, "true")
+    TestUtils.createTopicWithAdmin(admin, testTopicName, brokers, numPartitions, numReplicationFactor,
+      topicConfig = topicConfig)
+
+    // inherited local retention bytes is 1024
+    val configs = new util.HashMap[ConfigResource, util.Collection[AlterConfigOp]]()
+    configs.put(new ConfigResource(ConfigResource.Type.TOPIC, testTopicName),
+      util.Arrays.asList(
+        new AlterConfigOp(new ConfigEntry(TopicConfig.RETENTION_BYTES_CONFIG, "512"),
+          AlterConfigOp.OpType.SET),
+      ))
+    assertThrowsException(classOf[InvalidConfigurationException],
+      () => admin.incrementalAlterConfigs(configs).all().get(), "Invalid local retention size")
+  }
+
+  private def assertThrowsException(exceptionType: Class[_ <: Throwable],
+                                    executable: Executable,
+                                    message: String = ""): Throwable = {
+    assertThrows(exceptionType, () => {
+      try {
+        executable.execute()
+      } catch {
+        case e: ExecutionException => throw e.getCause
+      }
+    }, message)
+  }
+
+  private def verifyRemoteLogTopicConfigs(topicConfig: Properties): Unit = {
+    val logBuffer = brokers.flatMap(_.logManager.getLog(new TopicPartition(testTopicName, 0)))
+    assertTrue(logBuffer.nonEmpty)
```

Comment:
@divijvaidya 

The test fails at this place. Do you know how to fix it? Not sure why the log will be empty:

```
at kafka.admin.RemoteTopicCrudTest.verifyRemoteLogTopicConfigs(RemoteTopicCrudTest.scala:287)
```

Reply:
Please add the following line to TestUtils.createTopicWithAdmin(). This is because we currently only validate that metadata is consistent but after receiving metadata a broker has to process the metadata change as well which is async. Adding the following will ensure that `createTopicWithAdmin` only returns when log for the topic has indeed been created.

`TestUtils.waitUntilTrue(() => logManager.getLog(topicPartition).isDefined, "$testTopicName should be created")`